src/snap.py

```diff
@@ -140,13 +139,23 @@ def get_service_pid(self) -> int:
         Raises:
             SnapError if error occurs or if no pid string found in most recent log
         """
-        last_log = self.kafka.logs(services=[self.SNAP_SERVICE], num_lines=1)
-        pid_string = re.search(rf"{SNAP_NAME}.{self.SNAP_SERVICE}\[([0-9]+)\]", last_log)
+        java_processes = subprocess.check_output(
+            "pidof java", stderr=subprocess.PIPE, universal_newlines=True, shell=True
+        )
 
-        if not pid_string:
-            raise snap.SnapError("pid not found in snap logs")
+        logger.debug(f"Java processes: {java_processes}")
 
-        return int(pid_string[1])
+        for pid in java_processes.split():
+            with open(f"/proc/{pid}/cgroup", "r") as fid:
+                content = "".join(fid.readlines())
+
+                if f"{self.SNAP_NAME}.{self.SNAP_SERVICE}" in content:
+                    logger.info(
+                        f"Found Snap service {self.SNAP_SERVICE} for {self.SNAP_NAME} with PID {pid}"
+                    )
+                    return int(pid)
+
+        raise snap.SnapError(f"Snap {self.SNAP_NAME} pid not found")
```

Comment:
**question:** Although the change seems fine, what motivated it?

**todo:** Drop the logging level on L153 to `DEBUG`, no need for it in `INFO`.

Reply:
Right now we are not logging things out to console anymore (to avoid duplication of logs), and therefore this does not go into the syslogs (and as a result snap logs) anymore. So I had to slightly re-write the logic here, that was parsing the syslogs output to figure out the PID

Fine for the logging level, I don't have strong optinions about this

Reply:
AAAH, of course we are, makes sense then!